cdap-master/src/main/java/co/cask/cdap/data/tools/MetricsKafkaUpgrader.java

```diff
@@ -0,0 +1,167 @@
+    this.hConf = hConf;
+    this.hBaseTableUtil = hBaseTableUtil;
+    this.dsFramework = dsFramework;
+    this.kafkaTableName =  cConf.get(MetricsConstants.ConfigKeys.KAFKA_META_TABLE,
+                                     MetricsConstants.DEFAULT_KAFKA_META_TABLE);
+  }
+
+  private MetricsTable getOrCreateKafkaTable(String tableName, DatasetProperties props) {
+    MetricsTable table = null;
+    // old kafka table is in the default namespace
+    Id.DatasetInstance metricsDatasetInstanceId = Id.DatasetInstance.from(Constants.DEFAULT_NAMESPACE_ID, tableName);
+    try {
+      table = DatasetsUtil.getOrCreateDataset(dsFramework, metricsDatasetInstanceId,
+                                              MetricsTable.class.getName(), props, null, null);
+    } catch (Exception e) {
+      LOG.error("Exception while creating table {}.", tableName, e);
+    }
+    return table;
+  }
+
+  @Override
+  public void upgrade() throws Exception {
+    KafkaConsumerMetaTable kafkaMetaTableDestination =
+      new DefaultMetricDatasetFactory(cConf, dsFramework).createKafkaConsumerMeta();
+    // copy kafka offset from old table to new kafka metrics tabl
+    String kafkaTableNameOld = Joiner.on(".").join(Constants.SYSTEM_NAMESPACE, "default", kafkaTableName);
+    try {
+      // assuming we are migrating from 2.6 or 2.7
+      HTable hTable = getHTable(kafkaTableNameOld);
+      byte[] columnFamily = getColumnFamily(kafkaTableNameOld);
+
+      // iterate old table and copy all rows to the new kafkaConsumerMetaTable
+      try {
+        LOG.info("Starting upgrade for table {}", Bytes.toString(hTable.getTableName()));
+        Scan scan = getScan(columnFamily);
+        ResultScanner resultScanner = hTable.getScanner(scan);
+        Result result;
+        try {
+          while ((result = resultScanner.next()) != null) {
+            TopicPartition topicPartition = getTopicPartition(result.getRow());
+            if (topicPartition != null) {
+              long value  = Bytes.toLong(result.getFamilyMap(columnFamily).get(OFFSET_COLUMN));
+              kafkaMetaTableDestination.save(ImmutableMap.of(topicPartition, value));
+            } else {
+              LOG.warn("Invalid topic partition found {}", Bytes.toStringBinary(result.getRow()));
+            }
+          }
+          LOG.info("Successfully completed upgrade for table {}", Bytes.toString(hTable.getTableName()));
+        } finally {
+          resultScanner.close();
+        }
+      } catch (Exception e) {
+        LOG.info("Exception during upgrading metrics-kafka table {}", e);
+        throw Throwables.propagate(e);
+      } finally {
+        hTable.close();
+      }
+    } catch (IOException e) {
+      LOG.info("Unable to find table {}", kafkaTableNameOld);
+    }
+  }
+
+  private byte[] getColumnFamily(String tableName) throws DatasetManagementException {
+    // add old table for getting dataset spec
+    getOrCreateKafkaTable(tableName, DatasetProperties.EMPTY);
```

Comment:
You don't use the return value and since the method is just private and only called from here you can make it return void